executor/insert_common.go

```diff
@@ -778,18 +792,36 @@ func insertRowsFromSelectWorker(ctx context.Context, base insertCommon, batchSiz
 		case <-ctx.Done():
 			return nil
 		}
-		rows = make([][]types.Datum, 0, chk.NumRows())
+		rows = rows[:0]
+		rowsCacheIndex = 0
 		iter := chunk.NewIterator4Chunk(chk)
 		for innerChunkRow := iter.Begin(); innerChunkRow != iter.End(); innerChunkRow = iter.Next() {
-			innerRow := innerChunkRow.GetDatumRow(fields)
+			length := innerChunkRow.Len()
+			if len(datumRow) < length {
+				diff := length - len(datumRow)
+				for i := 0; i < diff; i++ {
+					datumRow = append(datumRow, types.Datum{}) //nolint:makezero
```

Comment:
Is it necessary to set
```
for i := 0; i < len(datumRow); i++ {
    datumRow[i] = types.Datum{}
}
```
like the else branch?

Reply:
Addressed